chore: do not run tests on transformed individual packages by default

BenChaimberg opened the pull request:
Because individual packages (packages that are release separately from `aws-cdk-lib`) are transformed and copied after being built and tested in their original location, there is no need to re-test them in their end location. Further, this can cause subtle issues such as changes in asset hashes due to modification of custom resource provider code during transformation. This change switches the default behaviour of the transform script to skip tests.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 198571b811a472f7d775b2f8a3b16fa39b8994a2
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c5156e90-7fc3-4b2c-961d-254d8f8cf0a1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

nija-at commented on scripts/transform.sh:
```diff
@@ -24,17 +24,17 @@ createSymlinks() {
 
 runtarget="build"
-run_tests="true"
+run_tests="false"
```
can you add a comment why we don't run tests by default?

just so someone doesn't come a month from now and change this back.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: ebabca2af60f5bf2f51c52bb30dcfc595dbbc2d0
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=64317ce3-a4f6-4610-affe-587af9f02b34%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

skinny85 commented:
One note about this PR - we have a way to ignore asset changes in our integration tests. ([example](https://github.com/aws/aws-cdk/blob/9bbfd185c2383612e2be7317a091b72cc5e7a120/packages/%40aws-cdk/aws-lambda-nodejs/test/integ.dependencies.ts#L1-L1)). Could this be a better approach?

I'm a little worried about not running unit and snapshot tests for packages that we will be publishing for customers to consume directly.

BenChaimberg commented:
The problem with using the pragma is that there are certain templates that contain multiple assets, where one of the assets comes from an external module but other assets are provided from the module. In the first case, the integration test *should not* care when the asset content changes but, in the second case, the integration test *should* care. Given how the current process works to remove the asset hashes, I don't see an immediate good way of working around this.

Tests will still be run on the modules in their pre-transformed state. This is similar to what we do for the sub-modules in `aws-cdk-lib`; tests are not run after transformation.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 2603bc5cd6a53225ee892833fa8d0342221f13d4
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1bd7a91f-ef44-4fbc-b312-9f64decf2851%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 1bb43c573511eae37abc45e168632403ac4e0d98
* Result: FAILED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0ff273ad-b757-4b76-8b07-44ac599913f0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

madeline-k commented:
@BenChaimberg I am a bit confused about what the issue is with integration tests that you're describing. But, why would we not want to run the unit tests? After transformation, the code will not be the same. So, I think the unit tests should still be run on the generated packages.

BenChaimberg commented:
<details>
<summary>Here's an example of a failed integration test caused by transformation:</summary>

```
Verifying integ.redshift-table.js against integ.redshift-table.expected.json ... CHANGED.
...
Resources
[~] AWS::Lambda::Function TableProviderframeworkonEvent97F3951A
 └─ [~] Code
     ├─ [~] .S3Bucket:
     │   └─ [~] .Ref:
     │       ├─ [-] AssetParametersdaeb79e3cee39c9b902dc0d5c780223e227ed573ea60976252947adab5fb2be1S3BucketDC4B98B1
     │       └─ [+] AssetParametersda69a9db84212fef7d9ed7767fc3096309e8b3c4aec374b40ddb76458acd5e24S3BucketD0ED29ED
     └─ [~] .S3Key:
         └─ [~] .Fn::Join:
             └─ @@ -8,7 +8,7 @@
                [ ]   "Fn::Split": [
                [ ]     "\|\|",
                [ ]     {
                [-]       "Ref": "AssetParametersdaeb79e3cee39c9b902dc0d5c780223e227ed573ea60976252947adab5fb2be1S3VersionKeyA495226F"
                [+]       "Ref": "AssetParametersda69a9db84212fef7d9ed7767fc3096309e8b3c4aec374b40ddb76458acd5e24S3VersionKey9972382C"
                [ ]     }
                [ ]   ]
                [ ] }
                @@ -21,7 +21,7 @@
                [ ]   "Fn::Split": [
                [ ]     "\|\|",
                [ ]     {
                [-]       "Ref": "AssetParametersdaeb79e3cee39c9b902dc0d5c780223e227ed573ea60976252947adab5fb2be1S3VersionKeyA495226F"
                [+]       "Ref": "AssetParametersda69a9db84212fef7d9ed7767fc3096309e8b3c4aec374b40ddb76458acd5e24S3VersionKey9972382C"
                [ ]     }
                [ ]   ]
                [ ] }
```

</details>

This is because the expected asset hash is calculated based on the original content of the custom resource provider as it existed in `@aws-cdk/aws-redshift`. However, when the alpha modules are transformed into their `-alpha` versions, there are content changes (mostly formatting) and so when the integration tests are re-synthesized, the asset hash is different from what was previously calculated.

From what I understand, the transformation process should not change the logic of the packages, just some rewriting of imports. So the transformation should not affect whether or not tests pass, but it could affect whether or not compilation succeeds. Essentially, I think there is low risk of tests failing that I think is outweighed by the benefit removing them – greater safety in the original tests.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildProject89A8053A-LhjRyN9kxr8o
* Commit ID: 2bb530767d1f37d9831cedaac7008fb73cd7272d
* Result: SUCCEEDED
* [Build Logs](https://xuogv3vm9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bd5790bd-a634-451d-ad99-d2d625b70885%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra commented:
This PR has been deemed to be abandoned, and will be closed. Please create a new PR for these changes if you think this decision has been made in error.
